include/no_os_spi.h

```diff
@@ -233,6 +253,10 @@ int32_t no_os_spi_init(struct no_os_spi_desc **desc,
 /* Free the resources allocated by no_os_spi_init(). */
 int32_t no_os_spi_remove(struct no_os_spi_desc *desc);
 
+/* Configure the SPI peripheral */
+int32_t no_os_spi_config(struct no_os_spi_desc *desc,
+			 const struct no_os_spi_init_param *param);
+
```

Comment:
Can you add more details about what this new function does and why is it needed? "Configure the SPI peripheral" is too vague and it doesn't look like you can actually reconfigure the fields defined in `struct no_os_spi_desc` (only those for xspi).